### ISSUE TITLE
`mksnapshot` binary for `armv7l` now works natively on `armv7l` machines, but still doesn't generate proper shapshots

### ISSUE THREAD

**Issue by ghost** [electron/electron#15312]
If you try to launch any application (Atom, for example) with `snapshot_blob.bin` which is generated by `mksnapshot` binary for `armv7l` architecture natively on `armv7l` machine it will throw an error and immediately exit:
```
#
# Fatal error in ../../v8/src/snapshot/deserializer.cc, line 79
# Check failed: magic_number_ == SerializedData::ComputeMagicNumber(external_reference_table_).
#
Illegal instruction
```
However, if you try to launch any application with `snapshot_blob.bin` which is generated by `mksnapshot` binary for `armv7l` architecture but on `amd64` machine it will be successfully launched.

I've also tried to play with arguments like `--no-use_ic` and `--arm_arch` but it still doesn't generate proper snapshots natively on `armv7l` machine.

Tested only with `mksnapshot` binary which is provided with `Electron 2.0.11`.

**Comment by visleaf** [comment]
I've also noticed that `mksnapshot`'s headers slightly differs from each other:
```
visleaf@workstation:~/Downloads/mksnapshot-v2.0.11-linux-armv7l-x64 $ ./mksnapshot --help
target simulator arm v7 neon hard
ARMv8=1 ARMv7=1 VFPv3=1 VFP32DREGS=1 NEON=1 SUDIV=1 USE_EABI_HARDFLOAT=1
...
```
```
pi@raspberry:~/Downloads/mksnapshot-v2.0.11-linux-armv7l $ ./mksnapshot --help
target arm v7 neon thumb hard
ARMv8=0 ARMv7=1 VFPv3=1 VFP32DREGS=1 NEON=1 SUDIV=1 USE_EABI_HARDFLOAT=1
...
```

I think that only @jkleinsc can help with `mksnapshot`-related problems, so bringing in him.

**Comment by jkleinsc** [comment]
Fixed in 3-0-x via #16041
